### ISSUE TITLE
Missing warning for `StringBuilder.Insert(int, String)` when the passed string contains only a single char

### ISSUE THREAD

**Issue by KvanTTT** [dotnet/roslyn-analyzers#7779]
```cs
int TestIndexOf(String s) => s.IndexOf("c", StringComparison.InvariantCulture); // CA1865
int TestLastIndexOf(String s) => s.LastIndexOf("d", StringComparison.InvariantCulture); // CA1865
bool TestStartsWith(String s) => s.StartsWith("a"); // CA1866
bool TestEndsWith(String s) => s.EndsWith("b"); // CA1866
bool Contains(String s) => s.Contains("c"); // CA1847

void TestAppend(StringBuilder stringBuilder) => stringBuilder.Append("e"); // CA1834
void TestInsert(StringBuilder stringBuilder) => stringBuilder.Insert(0, "f"); // Missing warning, but `Insert(int, char)` overload exists.
```

I find it's a bit inconsistent to have warning [CA1834](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1834) on `Append` call but no warning on `Insert` call.

*Background: I've found the issue while investigating an issue with missing warnings on similar call sites in Kotlin code* (https://youtrack.jetbrains.com/issue/KTIJ-36379/Add-inspection-to-suggest-char-overloads-for-startsWith-endsWith-and-indexOf-lastIndexOf-functions-from-STDLIB)
